src/mocompiler.js

```diff
@@ -1,28 +1,29 @@
 import { Buffer } from 'safe-buffer';
 import encoding from 'encoding';
-import { HEADERS, formatCharset, generateHeader, compareMsgid } from './shared.js';
+import { compareMsgid, formatCharset, generateHeader, HEADERS } from './shared.js';
 import contentType from 'content-type';
 
 /**
  * Exposes general compiler function. Takes a translation
  * object as a parameter and returns binary MO object
  *
- * @param {Object} table Translation object
+ * @param {import('./index.js').gettextTranslations} table Translation object
  * @return {Buffer} Compiled binary MO object
  */
 export default function (table) {
   const compiler = new Compiler(table);
 
   return compiler.compile();
-};
+}
 
 /**
  * Creates a MO compiler object.
  *
  * @constructor
- * @param {Object} table Translation table as defined in the README
+ * @param {import('./index.js').gettextTranslations} table Translation table as defined in the README
+ * @return {import('./index.js').Compiler} Compiler
  */
-function Compiler (table = {}) {
```

Comment:
this change is required because the null type check of typescript (a test will fail without)

Reply:
Yes, that's great! I would suggest to include them in the pre-TS PR.